### PR TITLE
Add interface objects to dbus

### PR DESCRIPTION
**Pull request by jk-ozlabs**
This PR continues the work from https://github.com/CodeConstruct/mctp/pull/39.

I have rebased @ThuBaNguyen 's commit to current `main`, so we have a single commit to implement the dbus change.

### CODE REVIEW

File: docs/mctpd.md [hunk 1]
```diff
@@ -104,3 +104,17 @@ busctl call au.com.codeconstruct.MCTP1 \
 
 Removes the MCTP endpoint from `mctpd`, and deletes routes and neighbour entries.
 
+## D-Bus /au/com/codeconstruct/mctp1/interfaces/<name>
+
+`mctpd` provides a D-Bus path of `/au/com/codeconstruct/mctp1/interfaces`.
+For each known MCTP interfaces, `mctpd` will populate an D-Bus object
+`/au/com/codeconstruct/mctp1/interfaces/<name>`. The D-Bus objects have
+interface `au.com.CodeConstruct.MCTP.Link1`.
```
Comment (jk-ozlabs) [inline docs/mctpd.md:112]: So we have a small inconsistency between `interfaces` and `Link` there. Can we unify?

**Comment by jk-ozlabs** [comment]
@ThuBaNguyen : Feel free to create this as a new PR on a branch in your own repo, if you prefer.

File: docs/mctpd.md (hunk 1 above)
Comment (ThuBaNguyen) [reply docs/mctpd.md:112]: We concerned about `/au/com/codeconstruct/mctp1/interfaces/<name>` and `/au/com/codeconstruct/mctp1/links/<name>` before. We can use `/au/com/codeconstruct/mctp1/links/<name>` for unifying.

Comment (jk-ozlabs) [reply docs/mctpd.md:112]: Yeah, we had discussed this previously, and settled on `/interfaces` (as you have implemented here). However, the dbus interface name still uses `Link`.

Comment (ThuBaNguyen) [reply docs/mctpd.md:112]: So what do you want? Change the object path to `/au/com/codeconstruct/mctp1/links/<name>`?

Comment (jk-ozlabs) [reply docs/mctpd.md:112]: no, we should have the interface name be `au.com.codeconstruct.MCTP.Interface1`. The object paths are good as they are.

Comment (jk-ozlabs) [reply docs/mctpd.md:112]: (I'm happy to do that if you prefer, but want to make sure I'm not stepping on your toes by doing so!)

Comment (ThuBaNguyen) [reply docs/mctpd.md:112]: Thanks. Please help me do that. So we can reuse this PR.

Comment (jk-ozlabs) [reply docs/mctpd.md]: OK, I have added: https://github.com/CodeConstruct/mctp/pull/44/commits/fabb82973cd25bc61d016669f07e94f76df734b4

If that looks good to you, I'll squash that into your change.

Comment (ThuBaNguyen) [reply docs/mctpd.md]: > OK, I have added: [fabb829](https://github.com/CodeConstruct/mctp/commit/fabb82973cd25bc61d016669f07e94f76df734b4)

The commit LGTM.

> 
> If that looks good to you, I'll squash that into your change.

**Pull request merged**